File: sdks/python/src/boxsdk/models/shared/task.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import user_mini as shared_user_mini
from ..shared import file_mini as shared_file_mini
from ..shared import taskassignments as shared_taskassignments

class TaskActionEnum(str, Enum):
    REVIEW = "review"
    COMPLETE = "complete"

class TaskCompletionRuleEnum(str, Enum):
    ALL_ASSIGNEES = "all_assignees"
    ANY_ASSIGNEE = "any_assignee"

class TaskTypeEnum(str, Enum):
    TASK = "task"


@dataclass_json
@dataclasses.dataclass
class Task:
    r"""Task
    A task allows for file-centric workflows within Box. Users can
    create tasks on files and assign them to other users for them to complete the
    tasks.
    """
    
    action: Optional[TaskActionEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('action') }})
    completion_rule: Optional[TaskCompletionRuleEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('completion_rule') }})
    created_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    created_by: Optional[shared_user_mini.UserMini] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_by') }})
    due_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('due_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    is_completed: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_completed') }})
    item: Optional[shared_file_mini.FileMini] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('item') }})
    message: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('message') }})
    task_assignment_collection: Optional[shared_taskassignments.TaskAssignments] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('task_assignment_collection') }})
    type: Optional[TaskTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    

File: ts-sdk/src/sdk/models/shared/task.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { UserMini } from "./usermini";
import { FileMini } from "./filemini";
import { TaskAssignments } from "./taskassignments";


export enum TaskActionEnum {
    Review = "review",
    Complete = "complete"
}

export enum TaskCompletionRuleEnum {
    AllAssignees = "all_assignees",
    AnyAssignee = "any_assignee"
}

export enum TaskTypeEnum {
    Task = "task"
}


// Task
/** 
 * A task allows for file-centric workflows within Box. Users can
 * create tasks on files and assign them to other users for them to complete the
 * tasks.
**/
export class Task extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=action" })
  action?: TaskActionEnum;

  @SpeakeasyMetadata({ data: "json, name=completion_rule" })
  completionRule?: TaskCompletionRuleEnum;

  @SpeakeasyMetadata({ data: "json, name=created_at" })
  createdAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=created_by" })
  createdBy?: UserMini;

  @SpeakeasyMetadata({ data: "json, name=due_at" })
  dueAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=is_completed" })
  isCompleted?: boolean;

  @SpeakeasyMetadata({ data: "json, name=item" })
  item?: FileMini;

  @SpeakeasyMetadata({ data: "json, name=message" })
  message?: string;

  @SpeakeasyMetadata({ data: "json, name=task_assignment_collection" })
  taskAssignmentCollection?: TaskAssignments;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: TaskTypeEnum;
}
